packages/modular-scripts/src/typecheck.ts

```diff
@@ -0,0 +1,13 @@
+import isCI from 'is-ci';
+import execa from 'execa';
+import getModularRoot from './utils/getModularRoot';
+
+export async function typecheck(): Promise<void> {
+  const options = ['--pretty'];
+  if (isCI) {
+    options.push('--noEmit');
+  }
+  await execa('tsc', options, {
```

Comment:
Can we do this programatically like we do for the `build` of packages at the moment?

Reply:
Done